### PR TITLE
ddl:support check invalid charset when create database

### PR DESCRIPTION
**Pull request by ciscoxll**
support check invalid charset when create database and create table 
respond issues #4486 

MySQL:

```
mysql> create database test1 character set uft8;
ERROR 1115 (42000): Unknown character set: 'uft8'

mysql> create table t(a int) character set uft8;
ERROR 1115 (42000): Unknown character set: 'uft8'
```

TiDB:
````
tidb> create database test1 character set uft8;
ERROR 1115 (42000): Unknown character set: 'uft8'

tidb> create table t(a int) character set uft8;
ERROR 1115 (42000): Unknown character set: 'uft8'

### CODE REVIEW

**Comment by ciscoxll** [comment]
@zimulala @winkyao PTAL

File: ddl/ddl_db_test.go [hunk 1]
```diff
@@ -124,6 +124,8 @@ func (s *testDBSuite) TestMySQLErrorCode(c *C) {
 	s.testErrorCode(c, sql, tmysql.ErrTooLongIdent)
 	sql = "create database test"
 	s.testErrorCode(c, sql, tmysql.ErrDBCreateExists)
+	sql = "create database test1 character set uft8;"
```
Comment (winkyao) [inline ddl/ddl_db_test.go]: Please add more test cases. More invalid charset and valid charset test.

File: infoschema/infoschema.go [hunk 2]
```diff
@@ -279,13 +281,14 @@ const (
...
-	codeMultiplePriKey  = 1068
-	codeTooManyKeyParts = 1070
+	codeDatabaseExists      = 1007
+	codeTableExists         = 1050
+	codeBadTable            = 1051
+	codeColumnExists        = 1060
+	codeIndexExists         = 1831
+	codeMultiplePriKey      = 1068
+	codeTooManyKeyParts     = 1070
+	codeUnknownCharacterSet = 1115
```
Comment (winkyao) [inline infoschema/infoschema.go]: Replace this by `codeUnknownCharacterSet = mysql.ErrUnknownCharacterSet`, By the way, you can modidy all above errCode to use mysqlErrxxx

**Comment by ciscoxll** [comment]
@zimulala @winkyao PTAL

File: ddl/ddl_api.go [hunk 3]
```diff
@@ -56,7 +56,12 @@ func (d *ddl) CreateSchema(ctx sessionctx.Context, schema model.CIStr, charsetIn
 	dbInfo := &model.DBInfo{
 		Name: schema,
 	}
+
 	if charsetInfo != nil {
+		err = checkCharsetAndCollate(charsetInfo.Chs, charsetInfo.Col)
+		if err != nil {
+			return err
```
Comment (zimulala) [inline ddl/ddl_api.go]: s/err/errors.Trace(err)

**Comment by zimulala** [comment]
```
mysql> create table t(a int) character set uft8;
ERROR 1115 (42000): Unknown character set: 'uft8'
tidb> create table t(a int) character set uft8;
Query OK, 0 rows affected (0.12 sec)
```
Could we fix it too?

**Comment by ciscoxll** [comment]
@zimulala @winkyao PTAL

**Pull request merged**